src/sa.h

```diff
@@ -0,0 +1,112 @@
+#include <stdio.h>
+#include <string.h>
+#include <malloc.h>
+#include "parsers/simple-fasta-parser.h"
+#include "parsers/simple-fastq-parser.h"
+#include <stdint-gcc.h>
+#include "helper.h"
+
+int *constructSARadix(struct Fasta fasta) {
+    char* x = fasta.fasta_sequence;
+    int n = fasta.fasta_len;
+    int* sa = malloc(n * sizeof *sa);
+    int* saCopy = malloc(n * sizeof *sa);
+    int** currentSa = &sa;
+    int** otherSa = &saCopy;
+    int** temp;
+
+    for (int i = 0; i<n; i++) {
+        sa[i] = i;
+    }
+    int* bucketsIndecies = fasta.alphabet.sightings;
+    int accumSum = 0;
+    for(int i=0; i<fasta.alphabet.size; i++) {
+        int sighting = bucketsIndecies[i];
+        bucketsIndecies[i] = accumSum;
+        accumSum += sighting;
+    }
+    int* buckets = malloc(fasta.alphabet.size*sizeof *buckets);
+
+    int charIndex;
+    char c;
+    int elemInBucket;
+    int saIndex;
+    for(int i=n-1; i>=0; i--) {
+        for(int j=0; j<fasta.alphabet.size; j++) buckets[j] = 0;
```

Comment:
You might consider `memset()` for this. It is standard compliant and you can set a buffer of any size to any integer.

Reply:
Thanks